src/kernels/jupyter/launcher/jupyterPasswordConnect.ts

```diff
@@ -482,7 +479,17 @@ export class JupyterPasswordConnect implements IJupyterPasswordConnect {
     // When URIs are removed from the server list also remove them from
     private onDidRemoveUris(uris: string[]) {
         uris.forEach((uri) => {
-            this.savedConnectInfo.delete(uri);
+            const newUrl = addTrailingSlash(uri);
+            this.savedConnectInfo.delete(newUrl);
```

Comment:
Don't usually like to do this, but snuck in a small related fix as the same time. Just something that I ran into fixing the other issue. Fixing that if you type in 'http://localhost:8888' it wouldn't clear the password (due to the trailing slash getting added).